webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -1892,71 +1892,32 @@ var userOverride = {
-                }
-                if (typeof events.onFaqClick === 'function') {
-                    window.kmWidgetEvents.onFaqClick = events.onFaqClick;
-                }
-                if (typeof events.onStartNewConversation === 'function') {
-                    window.kmWidgetEvents.onStartNewConversation =
-                        events.onStartNewConversation;
-                }
-                if (typeof events.onAttachmentClick === 'function') {
-                    window.kmWidgetEvents.onAttachmentClick =
-                        events.onAttachmentClick;
-                }
-                if (typeof events.onVoiceIconClick === 'function') {
-                    window.kmWidgetEvents.onVoiceIconClick =
-                        events.onVoiceIconClick;
-                }
-                if (typeof events.onShowResolvedClick === 'function') {
-                    window.kmWidgetEvents.onShowResolvedClick =
-                        events.onShowResolvedClick;
-                }
-                if (typeof events.onLocationIconClick === 'function') {
-                    window.kmWidgetEvents.onLocationIconClick =
-                        events.onLocationIconClick;
-                }
-                if (typeof events.onRateConversationClick === 'function') {
-                    window.kmWidgetEvents.onRateConversationClick =
-                        events.onRateConversationClick;
-                }
-                if (
-                    typeof events.onRateConversationEmoticonsClick ===
-                    'function'
-                ) {
-                    window.kmWidgetEvents.onRateConversationEmoticonsClick =
-                        events.onRateConversationEmoticonsClick;
-                }
-                if (typeof events.onSubmitRatingClick === 'function') {
-                    window.kmWidgetEvents.onSubmitRatingClick =
-                        events.onSubmitRatingClick;
-                }
-                if (typeof events.onRestartConversationClick === 'function') {
-                    window.kmWidgetEvents.onRestartConversationClick =
-                        events.onRestartConversationClick;
-                }
-                if (
-                    typeof events.onGreetingMessageNotificationClick ===
-                    'function'
-                ) {
-                    window.kmWidgetEvents.onGreetingMessageNotificationClick =
-                        events.onGreetingMessageNotificationClick;
-                }
-                if (typeof events.onNotificationClick === 'function') {
-                    window.kmWidgetEvents.onNotificationClick =
-                        events.onNotificationClick;
-                }
-                if (typeof events.onRichMessageButtonClick === 'function') {
-                    window.kmWidgetEvents.onRichMessageButtonClick =
-                        events.onRichMessageButtonClick;
-                }
+                Object.keys(events) &&
```

Comment:
is it possible to have something which sets it like:
window.kmWidgetEvents= events or something which extends only properties of events in kmWidgetEvents? is there any .extends something or something for that?

Reply:
can we use JQuery here?

Reply:
no